openshift/vault-manager.template.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: vault-manager
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    labels:
+      app: vault-manager
+    name: vault-manager
+  spec:
+    replicas: ${{REPLICAS}}
+    strategy:
+      type: RollingUpdate
+      rollingUpdate:
+        maxUnavailable: 0
+        maxSurge: 1
+    selector:
+      matchLabels:
+        app: vault-manager
+    template:
+      metadata:
+        labels:
+          app: vault-manager
+      spec:
+        containers:
+        - image: ${IMAGE}:${IMAGE_TAG}
+          imagePullPolicy: Always
+          name: vault-manager
+          command: ["/bin/sh"]
+          args: 
+          - "-c"
+          - "while true; do /vault-manager {{ if ${DRY_RUN} == 'True' }}--dry-run{{ end }}; sleep ${RECONCILE_SLEEP_TIME}; done;"
```

Comment:
Following Helm conditional conventions here. I haven't seen a conditional in a few other templates I checked out, so hoping this approach works.

Reply:
Nevermind, `oc process` does not support conditionals. Moving to a two template approach to support --dry-run.

Reply:
An alternative solution (probably better than two templates) is to set a DRY_RUN parameter with a default of : ""

```
- "while true; do /vault-manager $DRY_RUN; sleep ${RECONCILE_SLEEP_TIME}; done;"
...
parameters:
- name: DRY_RUN
  value: ""
  description: enables the dry-run cli flag when value is set to "--dry-run"
```

Reply:
you can also probably pass true/false to the flag? i also think it's `-dry-run` with a single dash?